broker-daemon/utils/create-http-server.js

```diff
@@ -30,24 +30,30 @@ function createHttpServer (protoPath, rpcAddress, { disableAuth = false, enableC
     app.use(corsMiddleware())
   }
 
+  // If the RPC address we use for daemon is set to a default route (0.0.0.0)
+  // then we want to make sure that we are instead making a request w/ grpc-gateway
+  // to local host since 0.0.0.0 would be an invalid address w/ the current cert
+  // setup
+  if (rpcAddress.includes('0.0.0.0')) {
+    rpcAddress = rpcAddress.replace('0.0.0.0', 'localhost')
+  }
+
   if (disableAuth) {
     app.use('/', grpcGateway([`/${protoPath}`], rpcAddress))
   } else {
     const key = fs.readFileSync(privKeyPath)
     const cert = fs.readFileSync(pubKeyPath)
     const channelCredentials = grpc.credentials.createSsl(cert)
 
-    // If the RPC address we use for daemon is set to a default route (0.0.0.0)
-    // then we want to make sure that we are instead making a request w/ grpc-gateway
-    // to local host since 0.0.0.0 would be an invalid address w/ the current cert
-    // setup
-    if (rpcAddress.includes('0.0.0.0')) {
-      rpcAddress = rpcAddress.replace('0.0.0.0', 'localhost')
-    }
-
     app.use('/', grpcGateway([`/${protoPath}`], rpcAddress, channelCredentials))
 
-    logger.debug(`Securing http RPC connections with TLS: key: ${privKeyPath}, cert: ${pubKeyPath}`)
+    // Handle 404s correctly for the server
```

Comment:
This handles 404s for `https` which was missing (we only handled 404s for http)

Reply:
should we have a single handler before we do the http/https divide?

Reply:
@treygriffith theres probably a way I can refactor this to make it nicer, but the `404` handling has to happen as the last action after grpc gateway, so it makes the code a little weird